nntrainer/src/tensor.cpp

```diff
@@ -770,4 +768,58 @@ Tensor Tensor::standardization() const {
 
   return result;
 }
+
+ExternalLoopInfo Tensor::computeExternalLoop(const Tensor &m) {
+  if (m.length() > this->length())
+    throw exception::not_supported("broadcasting *this is not supported");
+
+  const TensorDim m_dim = m.getDim();
+
+  ExternalLoopInfo e;
+
+  for (unsigned int i = 0; i < MAXDIM; ++i) {
+    if (m_dim.getTensorDim(i) == 1) {
+      continue;
+    }
+
+    if (dim.getTensorDim(i) == m_dim.getTensorDim(i)) {
+      e.strides[i] = m.strides[i];
+      continue;
+    }
+
+    std::stringstream ss;
+    ss << "[computeExternalLoop] broadcasting only allowed for"
+          "dimension value of 1 \n"
+       << "this: " << dim << "target: " << m_dim;
+    throw std::invalid_argument(ss.str().c_str());
+  }
+
+  /// calculate inner loop size
+  unsigned int inner_loop_size = dim.getTensorDim(3);
+
+  bool first_dim_none = m_dim.getTensorDim(3) == 1;
```

Comment:
I think its best to split this loop into multiple parts where each part has an objective.
Currently, this section does a lot of things, and not only bug prone but also hard to review.

Reply:
as discussed offline, lets keep this for now.